src/loader.js

```diff
@@ -146,15 +147,186 @@
   };
 
   // 4. Registry Objects
+  // For now, registry objects are a work in progress that don't fully integrate into the rest of the code base
 
   // 4.1.1
+  function createRegistry() {
+    var registry = new RegistryPrototype();
+    registry._registry = {};
+    if (__global.Map)
+      registry._registry.registryMap = new __global.Map();
```

Comment:
Are there any browsers that provide incomplete or non-functional implementations of Map? Or is it always safe to assume that everything else will work correctly here?